### PR TITLE
Start looking into getMapPos

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Been looking at this for reference as well.
https://github.com/OpenRCT2/OpenRCT2/commit/11b064617692b6a024571518166a9f5ed437d4ac

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Not sure if it's useful, but for reference: #197

**Review by AaronVanGeffen — APPROVED** [review]
I've played around with this for a bit, and viewport interaction still seems to work. The paint structs are obviously work in progress, but these are all nice improvements, I think.

File: src/OpenLoco/Paint/Paint.h [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        //uint32_t quadrantFrontIndex;                   // 0xE400C4
+        //const void* currentlyDrawnItem;                // 0xE4F0B4
+        //PaintEntry* endOfPaintStructArray;             // 0xE0C404
+        //PaintEntry* nextFreePaintStruct;               // 0xE0C40C
+        //Map::map_pos spritePosition;                   // 0xE3F090 and 0xE3F096
+        //PaintStruct* lastRootPS;                       // 0xE40120
+        //UI::ViewportInteraction::InteractionItem type; // 0xE3F0AC
+        //PaintStringStruct* paintStringHead;            // 0xE40118
+        //PaintStringStruct* lastPaintString;            // 0xE4011C
+        //Map::map_pos mapPosition;                      // 0xE3F0B0
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Paint/Paint.h:133]: Could you comment on why these have been commented out?

**Comment by duncanspumpkin** [comment]
Yeah I think I'll not remove my commented out part of the session for now its a nice little reference to help massage it into how OpenRCT2 works.

File: src/OpenLoco/Paint/Paint.h [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        //tunnel_entry rightTunnels[TUNNEL_MAX_COUNT];
+        //uint8_t rightTunnelCount;
+        //uint8_t verticalTunnelHeight;
+        //const Map::tile_element* surfaceElement;
+        //Map::tile_element* pathElementOnSameHeight;
+        //Map::tile_element* trackElementOnSameHeight;
+        //bool didPassSurface;
+        //uint8_t unk141E9DB;
+        //uint16_t waterHeight;
+        //uint32_t trackColours[4];
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Paint/Paint.h:154]: Should these be included, if they are not obviously present?

File: src/OpenLoco/Paint/Paint.h (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Paint/Paint.h:133]: These are all fields that I would like the session class to own. I have a little idea for how to handle this give me a sec.

File: src/OpenLoco/Paint/Paint.h (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Paint/Paint.h:157]: pretty sure the tunnels are included and I'm pretty sure the supports are also included. I'd rather keep them in for now then we can remove them as we go.

**Comment by AaronVanGeffen** [comment]
@duncanspumpkin This looks good to go as far as I'm concerned. I'll leave it to you to come up with a nice commit message. :wink:

**Pull request merged**